src/arena.c

```diff
@@ -2054,9 +2054,11 @@ arena_new(tsdn_t *tsdn, unsigned ind, extent_hooks_t *extent_hooks) {
 		/* background_thread_create() handles reentrancy internally. */
 		if (have_background_thread) {
 			bool err;
+			malloc_mutex_unlock(tsdn, &arenas_lock);
```

Comment:
Okay, can we dramatically reduce the range of code executed with ```arenas_lock``` held, so that we don't hold it by the time we create the background thread, and we therefore don't need to reacquire it?